src/components/Exchange/SwapBox.js

```diff
@@ -1922,6 +1922,11 @@ export default function SwapBox(props) {
                     className="Exchange-swap-input"
                     value={fromValue}
                     onChange={onFromValueChange}
+                    onKeyDown={e => {
+                      if (['e', 'E', '+', '-'].includes(e.key)) {
+                        e.preventDefault();
+                      }
+                    }}
```

Comment:
The logic in each input `onKeyDown` could be moved into its own function to prevent duplication